uui/components/datePickers/DatePicker.tsx

```diff
@@ -1,80 +1,153 @@
-                rawProps={ this.props.rawProps?.input }
-                id={ this.props.id }
+                cx={ cx(props.inputCx, isBodyOpen && uuiMod.focus) }
+                icon={ props.mode !== EditMode.CELL && systemIcons.calendar ? systemIcons.calendar : undefined }
+                iconPosition={ props.iconPosition || 'left' }
+                placeholder={ props.placeholder ? props.placeholder : format }
+                size={ props.size || '36' }
+                value={ inputValue || undefined }
+                onValueChange={ (v) => {
+                    setInputValue(v || '');
+                } }
+                onCancel={ () => {
+                    if (!props.disableClear && !!inputValue) {
+                        onValueChange(null);
+                    }
+                } }
+                isInvalid={ props.isInvalid }
+                isDisabled={ props.isDisabled }
+                isReadonly={ props.isReadonly }
+                tabIndex={ props.isReadonly || props.isDisabled ? -1 : 0 }
+                onFocus={ (e) => {
+                    setBodyIsOpen(true);
+                    props.onFocus?.(e);
+                } }
+                onBlur={ onBlur }
+                mode={ props.mode || defaultMode }
+                rawProps={ props.rawProps?.input }
+                id={ props.id }
             />
         );
     };
 
-    renderBody(props: DropdownBodyProps) {
+    const renderBody = (renderProps: DropdownBodyProps) => {
         return (
-            <DropdownContainer { ...props } focusLock={ false }>
+            <DropdownContainer
+                { ...renderProps }
+                focusLock={ false }
+            >
                 <DatePickerBody
-                    cx={ cx(this.props.bodyCx) }
-                    filter={ this.props.filter }
-                    value={ this.getValue() }
-                    setSelectedDate={ this.setSelectedDate }
-                    setDisplayedDateAndView={ this.setDisplayedDateAndView }
-                    changeIsOpen={ this.onToggle }
-                    renderDay={ this.props.renderDay }
-                    isHoliday={ this.props.isHoliday }
-                    rawProps={ this.props.rawProps?.body }
+                    value={ value }
+                    onValueChange={ onBodyValueChange }
+                    cx={ cx(props.bodyCx) }
+                    filter={ props.filter }
+                    isHoliday={ props.isHoliday }
+                    renderDay={ props.renderDay }
+                    rawProps={ props.rawProps?.body }
                 />
-                {this.props.renderFooter?.()}
+                {props.renderFooter?.()}
             </DropdownContainer>
         );
-    }
+    };
+
+    return (
+        <Dropdown
+            value={ isBodyOpen }
+            modifiers={ modifiers }
+            placement={ props.placement }
+            forwardedRef={ props.forwardedRef }
+            onValueChange={ (v) => {
```

Comment:
Just easier to debug

Reply:
But we shouldn't left code for debug in production. If we left an arrow function, it can worsen the perfomance and broke React.memo optimizations(we don't have them in Dropdown, but just like a best practice)